### ISSUE TITLE
remote method for calling data from react form

### ISSUE THREAD

**Issue by atherdon** [ChickenKyiv/recipe-search-api#53]
Based on Nick`s comment https://github.com/ChickenKyiv/recipe-search-react/issues/35#issuecomment-402366514

we need to have one API endpoint in order to populate our search fields

**Comment by chauhannishith** [comment]
I am unable to understand what is happening in this. Can you explain me what you have done?

**Comment by atherdon** [comment]
sure, no problem.
We should create a new API endpoint, that will return all data, that we need in our search form.
so, we'll create a new method at one of the models [sample1](https://github.com/ChickenKyiv/recipe-api-only/blob/master/server/models/recipe.js#L32-L95), [sample2](https://github.com/GroceriStar/groceristar/blob/master/server/models/department.js#L134-L173), [sample3](https://github.com/GroceriStar/groceristar/blob/master/server/models/grocery.js#L225) - sample3 is a crazy method, but i think it'll help you a bit in order to show how i manipulate the data

official docs here: https://loopback.io/doc/en/lb3/Remote-methods.html

**Comment by chauhannishith** [comment]
I think I didn't explain this properly. I meant to say what is happening currently in this repo, there is no readme so i have no idea. I tried reading the code and scripts but couldnt understand.

Also why do we need to create fetch when we are having a package which omits our hassle of server calls?

**Comment by atherdon** [comment]
@chauhannishith ohh, got it. I'll prepare an explanation and buzz you today or tomorrow

**Comment by chauhannishith** [comment]
Can you tell me some basics so I can start understanding this till you tell me in detail

**Comment by atherdon** [comment]
@chauhannishith i create a simple video for you, please take a look: https://vimeo.com/279601405

**Comment by chauhannishith** [comment]
can you clear out the table data related to this repository? There are multiple entries for each field

**Comment by chauhannishith** [comment]
![npm 19-07-2018 10_35_33](https://user-images.githubusercontent.com/29725515/42923070-3bcdfc20-8b41-11e8-9d1e-3c88877d97cc.png)

I am trying to get data from package into the DB but i am getting this error. Do you know where this auto update is? Also i deleted the existing data because there were multiple entries for everything

Sorry there are weather issues so the connectivity is still limited

**Comment by atherdon** [comment]
i get all your messages. just a bit busy. published 4 articles in 1 day :)
will reply soon, maybe when wake up, but 0 promises

**Comment by chauhannishith** [comment]
I still haven't solved this. Can you tell me how you used to run the import file? The database comes as undefined in the import file no matter what i do, I tried keeping the logic in root.js but i am still unable to do this

**Comment by atherdon** [comment]
Hi Nishith, yeah, i planned to get back to you asap. post this on pasue
while i'm not able to help you.


On 27-Jul-2018 at 05:32 AM, Nishith Chauhan wrote:


I still haven't solved this. Can you tell me how you used to run the import
file? The database comes as undefined in the import file no matter what i
do, I tried keeping the logic in root.js but i am still unable to do this

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-api/issues/53#issuecomment-408316051>,
or mute the thread

**Comment by atherdon** [comment]
@chauhannishith ok, i'm opened our server. will review it right now. more comments soon.
Btw, jjust to let you know - i've installed groceristar-fetch plugin. so we'll be able to make it work with you together too

**Comment by atherdon** [comment]
Ok, so looks like our database is clean right now. I start to understand more about your questions

**Comment by atherdon** [comment]
@chauhannishith it's not cool to say. but i think we should leave it as is. 
i tried to debug it, but what i find - that server doesn't return a datasource. you can check my latest commits and see what i've done. 
i'm a bit angry to loopback, because i think this issue related to their changes.
One month ago main server, related to groceristar project started to work strange. and i tried to fix it fast and then gave up and this issue forced me to start a process when we leave our old servers and just migrate to GraphQL approach. what do you think if i'll create a task for you, related to learning how GraphQL works and then we'll rebuild this server together?
I think issues from groceristar server and this server are linked, so maybe it's a good time to move to a new, more easier and clean way to do it.

**Comment by atherdon** [comment]
tell me what do you think

**Comment by atherdon** [comment]
Hold on. i think i find something

**Comment by atherdon** [comment]
Nah, it not works... and as i opened main repository - a lot of other issues people posting too. so this is a dead end i assume

**Comment by chauhannishith** [comment]
Hi @atherdon I think you haven't read my messages on slack. I won't be able to help you at least for some time right now in creating new stuff. But I think MLAB has a feature to export JSON files, you can search it and it might solve your issue for atleast some time right now.



> i tried to debug it, but what i find - that server doesn't return a datasource.

i also encountered the same issue and tried to work around in root.js file but didn't work.

**Comment by atherdon** [comment]
nope, i didn't saw your messages. i think slack should send notification
to email, when i'm not at workroom - but i never works fine.


On 30-Jul-2018 at 04:43 AM, Nishith Chauhan wrote:


Hi @atherdon <https://github.com/atherdon> I think you haven't read my
messages on slack. I won't be able to help you at least for some time right
now in creating new stuff. But I think MLAB has a feature to export JSON
files, you can search it and it might solve your issue for atleast some
time right now.

i tried to debug it, but what i find - that server doesn't return a
datasource.

i also encountered the same issue and tried to work around in root.js file
but didn't work.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-api/issues/53#issuecomment-408745343>,
or mute the thread
